bootstrapper-maven-plugin/src/main/resources/templates/Reconciler.java

```diff
@@ -0,0 +1,60 @@
+package {{groupId}};
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.javaoperatorsdk.operator.api.config.informer.InformerConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceInitializer;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.api.reconciler.ControllerConfiguration;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+
+import java.util.Map;
+import java.util.Optional;
+
+@ControllerConfiguration
+public class {{artifactClassId}}Reconciler implements Reconciler<{{artifactClassId}}CustomResource>, EventSourceInitializer<{{artifactClassId}}CustomResource> {
```

Comment:
I don't think we should be generating such complex code. While I understand the idea is to show examples of what can be done, in most instances, this code will only be deleted / changed and might confuse less advanced users who might think it is required for some reason. Generally speaking, I would expect code generation to only generate the minimal amount of code that is required to get started and/or things that will *always* be required (though you could make a point that things are always needed should be part of the framework anyway).

Reply:
We can add comment to explain, but see the idea above to generate a minimal working implementatio. As it is done in generators like quarkus, that generates an ednpoint with "hellow world" response.

Reply:
Well, a minimal implementation is simply a reconciler that does nothing or maybe just outputs the name of resources it watches when an event occurs on of them. That's not what this is doing as it's adding an informer for `ConfigMap`, which has nothing to do with reconciling the primary resource.

Reply:
I mean somthing meaningful that does something. I don't think this is bad, actually might help people to getting started. Also can add there comment, why is it this way.

Reply:
If we want to have something that does something out of the box, we should use the managed dependent approach, not this manual wiring thing.

Reply:
I think it is although eventually a preferred usage for some cases, it simply does not work for all cases. And the idea is to have an entry project.

Reply:
A basic project should use managed dependents, period. It should address most users' needs. If they need more than managed dependents, they're already past the basic stuff and don't need to have their hand held by the code generator.

Reply:
Also, generated code is not meant to be documentation, it's meant to get people started quickly without adding stuff they don't need just for the sake of exposure. Otherwise, that's a tutorial and the goals are different. The other aspect to take into account is that this generator will have to be maintained to follow the evolution of the rest of the code and the more complex it is, the more tricky it will be to maintain it up to date.

Reply:
Not sure if following your logic, the dependent resource usage is actually more code, additional class, etc. I think we could actually go with also that, so if you insist I can add a ConfigMapDependentResource. 
Eventually I want to add an Integration test (next PR) that checks if the ConfigMap changed / created. That is why the logic there. As usually generators do, to generate some basic tests also. 

This but also other generators like Spring Boot, Quarkus, also not meant as docs, this neiher. But it is very helpful (at least to my experience) to have something very simple that is actually working and does something, instead of empty code. Since much easier to modify that, then figuring out from docs how to start with simplest case. This is worth the maintenance burden.

So can change it to use dependent resource, and the test later.